### ISSUE TITLE
What is the best way to merge two Embedding Layers?

### ISSUE THREAD

**Issue by eromoe** [keras-team/keras#9485]
I train two word2vec model base on different features of a sentence. 
Such as a Chinese text can present as unicode char or pinyin(alphabet) . 

I think the training step would be : 

1. input sentence
1. convert sentence to char embedding( vocab_size 40000+, dim 200 ) and pinyin embedding(vacab_size 500+, dim 200).
1. I want to combine above two embeddings,  to  shape (input_lenght, 2, 200) 
1. then feed to lstm  (maybe add a cnn in front of lstm)

But I found that embedding layer only accept fixed (vocab_size, dim) .
```
char_em = Embedding(
    output_dim = char_weights.shape[1],
    input_dim = char_weights.shape[0], # vocab_size
    mask_zero = True,
    weights = [char_weights],
)
```

What is the best way to merge two Embeddings for my purpose ?

-----

I saw some solustion:

From https://github.com/keras-team/keras/issues/1675

```
src_model = Sequential()
src_model.add(Embedding(collapse_dim, embedding_size, input_length=1))
dst_model = Sequential()
dst_model.add(Embedding(collapse_dim, embedding_size, input_length=1))

encoder = Sequential()
encoder.add(Merge([src_model, dst_model], mode='concat'))
encoder.add(Reshape((embedding_size,)))#(None, 1, 128) -> (None, 128)
encoder.add(Dense(3*embedding_size, init='uniform'))
encoder.add(Activation('relu'))
```
I don't understand why it reshape (input_length, 1, 128) -> (input_length, 128) ,  doesn't the shape is  `(input_length, 2, 128)` ?

From https://github.com/keras-team/keras/issues/1675
```
left = Sequential()
left.add(LSTM(output_dim=600, input_shape=(80,600)))
left.add(Dense(600))

right = Sequential()
right.add(LSTM(output_dim=600, input_shape=(80,600)))
right.add(Dense(600))

model = Sequential()
model.add(Merge([left, right],mode='sum'))
model.add(Dense(4))
model.add(Activation('softmax'))
model.compile(loss='categorical_crossentropy',optimizer='rmsprop')

model.fit([X_left, X_right], Y_train, nb_epoch=100,validation_split=0.2,show_accuracy=True)
```

I can understand this, but I only want train one lstm layer,  I think it is much more efficient than two.

**Comment by abushoeb** [comment]
Check out this https://github.com/keras-team/keras/issues/1675

**Comment by agarnitin86** [comment]
Merge doesn't work in keras version 2.24
